Random UI test fail - 1 second timing issue

_Originally posted by @Steve-Mcl in https://github.com/FlowFuse/flowfuse/issues/3868#issuecomment-2115494842_
            
```
 FAIL  test/unit/frontend/utils/elapsedTime.spec.js > elapsedTime > compares relative to now if only a to date is passed
AssertionError: expected '30 minutes, 14 seconds' to be '30 minutes, 15 seconds' // Object.is equality

- Expected
+ Received

- 30 minutes, 15 seconds
+ 30 minutes, 14 seconds

 ❯ test/unit/frontend/utils/elapsedTime.spec.js:66:35
     64|         soon.setSeconds(soon.getSeconds() + 15)
     65| 
     66|         expect(elapsedTime(soon)).toBe('30 minutes, 15 seconds')
       |                                   ^
     67|     })
     68| 
```

--- a/test/unit/frontend/utils/elapsedTime.spec.js
+++ b/test/unit/frontend/utils/elapsedTime.spec.js
@@ -62,6 +62,9 @@ describe('elapsedTime', () => {
         const soon = new Date()
         soon.setMinutes(soon.getMinutes() + 30)
         soon.setSeconds(soon.getSeconds() + 15)
+        // This test would occasionally fail due to a combination of the random ms value in `soon`, the time taken to
+        // run the test and the fact that the elapsed time string is calculated using Math.floor
+        soon.setMilliseconds(soon.getMilliseconds() + 500) // Grace to prevent random test fails.
 
         expect(elapsedTime(soon)).toBe('30 minutes, 15 seconds')
     })

